modulus/models/module.py

```diff
@@ -394,3 +395,12 @@ def num_parameters(self) -> int:
         for name, param in self.named_parameters():
             count += param.numel()
         return count
+
+
+class OmegaConfEncoder(json.JSONEncoder):
```

Comment:
Why have a custom JSONEncoder instead of something like this before the json.dump call?
```python
self._args = {k : (
    omegaconf.OmegaConf.to_container(v, resolve = True)
    ) if isinstance(v, (omegaconf.ListConfig, omegaconf.DictConfig)) else v}
```

Reply:
In my opinion both are the same, but I think custom JSON encoder is a cleaner implementation and is consistent with this: https://github.com/python/cpython/blob/main/Lib/json/encoder.py#L161

Reply:
IMO, we should fix this at the call site instead of adding omegaconf as a dependency to modulus-core as mentioned in  https://github.com/NVIDIA/modulus-launch/pull/79#pullrequestreview-1579366654

Creating a custom JSON encoder for one package is clean. But I worry that we'll get into a maintenance issue if we need to start having more custom encoders which would then have to inherit from this `OmegaConfEncoder` and create a serial inheritance chain of encoders.

Reply:
I agree. Fixing this at the call site is a better approach. I will close this PR then.